docs/providers/splunk.md

```diff
@@ -0,0 +1,71 @@
+# Splunk
+
+## Backend
+
+Using the `splunk` backend class, you can query any metrics available in Splunk Enterprise to create an SLO.
```

Comment:
Actually, i'm unable to test for several reasons :
- Generating API Tokens requires admin access, which i lack
- The API by default enforces an IP whitelist (in which my test server is probably not)
- I'm using SSO so I cannot use my login/password

However it means adding a way to use APi token in the config would help making it possible for exveryone.

Anyway, the documentation clearly states the APIs are identical to Splunk Enterprise (onprem) & Cloud (although limited to mostly performing searches, which is what we need)
https://docs.splunk.com/Documentation/SplunkCloud/9.0.2303/RESTTUT/RESTandCloud

Reply:
I've requested an API token from our splunk management team, i'll test when I get it.
Anyway given the way i've coded it now, there will be a few changes required

Reply:
So I've tested the splunk cloud API using a Bearer token instead of login/password and results are equivalent. However, I haven't been able to test using the slo-generator for obscure network issues (their API is whitelisting IPs).

However, their code confirms the "splunkToken" argument to build the client is indeed used as a Bearer token :
https://github.com/splunk/splunk-sdk-python/blob/e44135872675d8f5450b7eef990735b233477b27/splunklib/binding.py#LL495C11-L495C11

And I've implemented token use in the code 
https://github.com/google/slo-generator/blob/b5a8189fab7e6cb2e67fcdc4bda1a73e5777e523/slo_generator/backends/splunk.py#L31-L36

So there's no reason for it not to work.